### PR TITLE
Fix success response code on jpeg images

### PR DESCRIPTION
**Pull request by AKalinich-Luxoft**
Fixes 
[#319](https://github.com/smartdevicelink/sdl_hmi/issues/319)
[#321](https://github.com/smartdevicelink/sdl_hmi/issues/321)
[#329](https://github.com/smartdevicelink/sdl_hmi/issues/329)

This PR is **ready** for review.

### Testing Plan
Manual testing

### Summary
Enhanced image validation function and added check for template non-png images.

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by jacobkeeler — CHANGES_REQUESTED** [review]
Also fix merge conflicts

File: app/model/sdl/Abstract/Model.js [hunk 1]
```diff
@@ -984,57 +984,69 @@ SDL.SDLModel = Em.Object.extend({
...
+    };
+
+    // Retain reference to requestID checklist as this element might be deleted
+    // by finalizeImageValidation() if all images are verified be template extension function
+    let checkList = this.imageCheckList[requestID];
+    for(var i = 0; checkList && i < checkList.length; i++) {
+      if (!is_valid_template_extension(checkList[i])) {
+        checkList[i].checkResult = {
+          code: false,
+          info : "Template image extension is not valid"
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply app/model/sdl/Abstract/Model.js]: @jacobkeeler fixed in https://github.com/smartdevicelink/sdl_hmi/pull/462/commits/fd93ebfeab939e56fec00794e5e44fef39b8a920

File: app/model/sdl/Abstract/Model.js [hunk 2]
```diff
@@ -984,57 +984,69 @@ SDL.SDLModel = Em.Object.extend({
...
-          var path = SDL.SDLModel.imageCheckList[requestID][i].path;
-          if(path === formattedImgPath) {
-            SDL.SDLModel.imageCheckList[requestID][i].checkResult = true;
-            break;
-          }
-        }
+      image.onload = function() {
+        checkList[this.checkIndex].checkResult = {
+          code: true,
+          info : null
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply app/model/sdl/Abstract/Model.js]: @jacobkeeler fixed in https://github.com/smartdevicelink/sdl_hmi/pull/462/commits/fd93ebfeab939e56fec00794e5e44fef39b8a920

File: app/model/sdl/Abstract/Model.js [hunk 3]
```diff
@@ -984,57 +984,69 @@ SDL.SDLModel = Em.Object.extend({
...
-          if(path === formattedImgPath) {
-            SDL.SDLModel.imageCheckList[requestID][i].checkResult = false;
-            break;
-          }
-        }
+
+      image.onerror = function() {
+        checkList[this.checkIndex].checkResult = {
+          code: false,
+          info : "Requested image(s) not found"
```
[the remark this answers is not in the document]
Comment (AKalinich-Luxoft) [reply app/model/sdl/Abstract/Model.js]: @jacobkeeler fixed in https://github.com/smartdevicelink/sdl_hmi/pull/462/commits/fd93ebfeab939e56fec00794e5e44fef39b8a920

**Comment by AKalinich-Luxoft** [comment]
> Also fix merge conflicts

@jacobkeeler conflicts have been resolved in https://github.com/smartdevicelink/sdl_hmi/pull/462/commits/8eabfcc4de5786be1cdc0460e9b764c855f5e4f7

**Review by jacobkeeler — APPROVED** [review]

**Pull request merged**